test/unit/spec/LoginRouter_spec.js

```diff
@@ -170,48 +170,75 @@ function (Okta, Q, Backbone, SharedUtil, CryptoUtil, Logger, OktaAuth, Util, Exp
       var fn = function () { setup({ foo: 'bla' }); };
-      expect(fn).not.toThrow(Errors.ConfigError);
+      expect(fn).not.toThrowError(Errors.ConfigError);
       expectUnexpectedFieldLog('foo');
     });
     it('has the correct error message if unknown option is passed as a widget param', function () {
-      spyOn(Logger, 'warn');
+      spyOn(CourageLogger, 'warn');
       var fn = function () { setup({ foo: 'bla' }); };
       expect(fn).not.toThrow();
       expectUnexpectedFieldLog('foo');
     });
-    it('throws a ConfigError if el is not passed as a widget param', function () {
+    it('logs a ConfigError error if el is not passed as a widget param', function () {
       var fn = function () { setup({ el: undefined }); };
-      expect(fn).toThrowError(Errors.ConfigError);
+      expect(fn).not.toThrow();
+      expect(Logger.warn).toHaveBeenCalled();
     });
     it('has the correct error message if el is not passed as a widget param', function () {
       var fn = function () { setup({ el: undefined }); };
-      expect(fn).toThrowError('"el" is a required widget parameter');
+      expect(fn).not.toThrow();
+      var err = Logger.warn.calls.mostRecent().args[0];
+      expect(err.name).toBe('CONFIG_ERROR');
+      expect(err.message).toEqual('"el" is a required widget parameter');
     });
-    it('throws a ConfigError if baseUrl is not passed as a widget param', function () {
+    it('logs a ConfigError if baseUrl is not passed as a widget param', function () {
       var fn = function () { setup({ baseUrl: undefined }); };
-      expect(fn).toThrowError(Errors.ConfigError);
+      expect(fn).not.toThrowError(Errors.ConfigError);
+      expect(Logger.warn).toHaveBeenCalled();
     });
     it('has the correct error message if baseUrl is not passed as a widget param', function () {
       var fn = function () { setup({ baseUrl: undefined }); };
-      expect(fn).toThrowError('"baseUrl" is a required widget parameter');
+      expect(fn).not.toThrow();
+      var err = Logger.warn.calls.mostRecent().args[0];
+      expect(err.name).toBe('CONFIG_ERROR');
+      expect(err.message).toEqual('"baseUrl" is a required widget parameter');
+    });
+    itp('uses a default globalSuccessFn if an undefined globalSuccessFn is passed as a widget param', function () {
+      return expectPrimaryAuthRender({ globalSuccessFn: undefined });
+    });
+    itp('uses a default globalSuccessFn if a null globalSuccessFn is passed as a widget param', function () {
+      return expectPrimaryAuthRender({ globalSuccessFn: null });
     });
-    it('throws a ConfigError if globalSuccessFn is not passed as a widget param', function () {
-      var fn = function () { setup({ globalSuccessFn: undefined }); };
-      expect(fn).toThrowError(Errors.ConfigError);
+    itp('uses a default globalErrorFn if an undefined globalErrorFn is passed as widget param', function () {
+      return expectPrimaryAuthRender({ globalErrorFn: undefined });
```

Comment:
I'm not sure that this test is actually testing that the default handler is used? This test feels more like "assert that things work if I omit the handler during setup" ?